code/modules/cooking/cooking_container.dm

```diff
@@ -0,0 +1,307 @@
+/obj/effect/cooking_container_lip
+	icon = 'icons/obj/cooking/containers.dmi'
+	icon_state = "blank"
+	vis_flags = VIS_INHERIT_ID
+
+/**
+ * Cooking containers are used in ovens, fryers and so on, to hold multiple
+ * ingredients for a recipe. They interact with the cooking process, and link
+ * up with the cooking code dynamically. Originally sourced from the Aurora,
+ * heavily retooled to actually work with PCWJ Holder for a portion of an
+ * incomplete meal, allows a cook to temporarily offload recipes to work on
+ * things factory-style, eliminating the need for 20 plates to get things done
+ * fast.
+ **/
+/obj/item/reagent_containers/cooking
+	icon = 'icons/obj/cooking/containers.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	volume = 240
+	var/datum/cooking/recipe_tracker/tracker = null
+	/// Icon state of the lip layer of the object
+	var/lip
+	var/obj/effect/cooking_container_lip/lip_effect
```

Comment:
don't forget to clean this (and its tracker, perhaps) in destroy.

Also, please reorder the declarations to put the existing variables before the new ones

Reply:
Done